nixos/roles/k3s/server.nix

```diff
@@ -304,12 +304,32 @@ in {
       ensureDatabases = [ "kubernetes" ];
       ensureUsers = [ {
         name = "root";
-        ensurePermissions = {
-          "DATABASE kubernetes" = "ALL PRIVILEGES";
-        };
       } ];
     };
 
+    systemd.services.kubernetes-ensure-db-permissions = {
+      description = "Ensure the root user has all permissions on kubernetes db";
+      wantedBy = [ "k3s.service" ];
+      requires = [ "postgresql.service" ];
+      script = ''
+        PSQL="${config.services.postgresql.package}/bin/psql --port=${toString config.services.postgresql.port}"
+        for i in {1..10}; do
+          $PSQL -d postgres -c "" 2> /dev/null && break
+          sleep 0.5
+
+          if [[ $i -eq 10 ]];then
+            echo "couldn't establish connection to postgres"
+            exit 1
+        fi
+        done
+          $PSQL -tAc 'GRANT ALL PRIVILEGES ON DATABASE kubernetes TO "root"'
```

Comment:
This will most likely also not work on PostgreSQL 15+. The user needs to have permissions on the `public` schema too or ownership of the DB. See https://github.com/NixOS/nixpkgs/issues/216989 for more information.